src/main/scala/scodec/codecs/TupleCodec.scala

```diff
@@ -3,14 +3,401 @@ package codecs
 
 import scodec.bits.BitVector
 
-private[scodec] final class TupleCodec[A, B](codecA: Codec[A], codecB: Codec[B]) extends Codec[(A, B)] {
+private[scodec] final class TupleCodec[A, B](A: Codec[A], B: Codec[B]) extends Codec[(A, B)] {
 
   override def encode(t: (A, B)) =
-    Codec.encodeBoth(codecA, codecB)(t._1, t._2)
+    Codec.encodeBoth(A, B)(t._1, t._2)
 
   override def decode(buffer: BitVector) =
-    Codec.decodeBoth(codecA, codecB)(buffer)
+    Codec.decodeBoth(A, B)(buffer)
 
-  override def toString = s"($codecA, $codecB)"
+  def ~~[C](C: Codec[C]): Tuple3Codec[A,B,C] = new Tuple3Codec(A,B,C)
+
+  override def toString = s"($A, $B)"
+}
+
+private[scodec] class Tuple3Codec[A,B,C](A: Codec[A],
+                                         B: Codec[B],
+                                         C: Codec[C]) extends Codec[(A,B,C)] {
+  def ~~[D](D: Codec[D]) = new Tuple4Codec(A,B,C,D)
+
+  override def decode(bits: BitVector) = {
+    for {
+      a <- A.decode(bits)
+      b <- B.decode(a.remainder)
+      c <- C.decode(b.remainder)
+    } yield DecodeResult((a.value,b.value,c.value), c.remainder)
+  }
+
+  override def encode(abc: (A,B,C)) =
+    for {
+      bits <- A.encode(abc._1)
+      bits2 <- B.encode(abc._2)
+      bits3 <- C.encode(abc._3)
+    } yield bits ++ bits2 ++ bits3
+
+  def widenOpt[X](to: (A,B,C) => X, from: X => Option[(A,B,C)]): Codec[X] = 
```

Comment:
Oh right... how about we pick another name then and in the implementation, delegate to `widenOpt`?